include/circt/Dialect/Verif/VerifOps.h

```diff
@@ -12,6 +12,7 @@
 #include "circt/Dialect/HW/HWDialect.h"
 #include "circt/Dialect/HW/HWTypes.h"
 #include "circt/Dialect/Verif/VerifDialect.h"
+#include "mlir/Interfaces/InferTypeOpInterface.h"
```

Comment:
Why does adding a FIRRTL op require adding headers to other dialects? Are these missing from the FIRRTL places, or unrelated?
(They appear necessary, but haven't looked at why, do you remember/know?)

Reply:
I really don't get this part of TableGen. The key here is the `include "mlir/Interfaces/InferTypeOpInterface.td"` line in `VerifOps.td`. If you have an op with something like
```
let results = (outs I1:$result);
```
TableGen will only produce the implicit conversion from Op to Value and the builder function without the result `TypeRange` if you have the `InferTypeOpInterface.td` include further up in the TD file. If you comment out the include, the accessors and builders with inferred result type *won't* be generated. Even if your op doesn't implement or inherit from that `InferTypeOpInterface` at all. All that's needed is the include. Blows my mind.